src/app/components/pages/admin-cell/admin-cell.component.ts

```diff
@@ -167,9 +182,109 @@ export class AdminCellComponent implements OnInit {
+
+    this.getCycles();
+    this.getTaskTypes();
+    this.eventForm = this.formBuilder.group(
+      {
+        nb_volunteers_needed: [0, [Validators.required, Validators.min(0)]],
+        nb_volunteers_standby_needed: [0, [Validators.min(0)]],
+        cell_id: [cellId, Validators.required],
+        cycle_id: [null, Validators.required],
+        task_type_id: [null, Validators.required],
+        start_date: [null, Validators.required],
+        end_date: [null, Validators.required]
+      },
+      {
+        validator: this.dateValidator()
+      }
+    );
+  }
+
+  getCycles(): void {
+    this.cycleService.getCycles()
+      .subscribe(cycles => this.cycles = cycles.results);
+  }
+
+  getTaskTypes(): void {
+    this.tasktypeService.getTasktypes()
+      .subscribe(taskTypes => this.tasktypes = taskTypes.results);
+  }
+
+  dateValidator() {
+    return (group: FormGroup) => {
+
+      const date_start = group.controls['start_date'];
+      const date_end = group.controls['end_date'];
+
+      if (date_start.value && date_end.value && date_start.value >= date_end.value) {
+        return date_end.setErrors({
+          dateEndBeforeDateStart: true
+        });
+      }
+
+      if (!date_start.value && date_end.value) {
+        return date_start.setErrors({
+          dateStartMissing: true
+        });
+      }
+
+      if (date_start.value && !date_end.value) {
+        return date_end.setErrors({
+          dateEndMissing: true
+        });
+      }
+    };
+  }
+
+  createEvent() {
+
+    if (this.eventForm.valid) {
+
+      this.eventService.createEvent(this.eventForm.value).subscribe(
         data => {
-          window.location.assign(this.cellService.getExportCellLink(data.export_link));
+          this.myModals.get('create event').toggle();
+          this.eventForm.reset();
+          this.notificationService.success('Création réussie',
+            `La plage horaire a été créé`);
+
+          this.get_events();
+        },
+        err => {
+
+          let errorMessage = '';
+          if (err.error.nb_volunteers_needed) {
+            errorMessage += `nb_volunteers_needed erreur: ${err.error.nb_volunteers_needed[0]},`;
+          }
+          if (err.error.cycle_id) {
+            errorMessage += `cycle_id erreur: ${err.error.cycle_id[0]},`;
+          }
+          if (err.error.task_type_id) {
+            errorMessage += `task_type_id erreur: ${err.error.task_type_id[0]},`;
+          }
+          if (err.error.start_date) {
+            errorMessage += `Start date erreur: ${err.error.start_date[0]},`;
+          }
+          if (err.error.end_date) {
+            errorMessage += `End date erreur: ${err.error.end_date[0]},`;
+          }
+          if (err.error.non_field_errors) {
+            this.eventForm.controls['end_date'].setErrors({
+              apiError: err.error.non_field_errors
+            });
+            errorMessage += `End date erreur: ${err.error.non_field_errors[0]},`;
```

Comment:
`non_field_errors` need to be put in a general information error on not in a specific field.

Actually you can have this kind of display.

![screenshot-2018-5-30 volontaria](https://user-images.githubusercontent.com/12053720/40722374-23045aa0-63ea-11e8-911a-20e460b404f0.png)

You can use a global variable `myVariableOfError: string[]` to take the content of the `non_field_error` and display it like this on the bottom of the form :

```html
<div class="alert alert--danger" *ngIf="myVariableOfError">
    <ul>
        <li *ngFor="let error of maVariableOfError">
            {{error}}
        </li>
    </ul>
</div>
```